OpenRA.Game/Traits/World/ScreenMap.cs

```diff
@@ -68,6 +71,16 @@ public void Add(Player viewer, FrozenActor fa)
 			partitionedFrozenActors[viewer].Add(fa, FrozenActorBounds(fa));
 		}
 
+		public void Add(IEffect effect, WPos position)
+		{
+			var bounds = effect.VisualBounds(worldRenderer);
+			if (bounds.Width > 0 && bounds.Height > 0)
+			{
+				var screenPos = worldRenderer.ScreenPxPosition(position);
```

Comment:
`screenPos` gives the middle of the sprite, but you treat it as the top-left below.  This will offset everything in the map by `(width, height) / 2`.